File: class23_homework/B-Bozhidar-Nikolov-5/JavaVersion/PassengerPlane/src/PassengerPlane.java
import java.util.Scanner;

public class PassengerPlane {
	
	private int capacity[][] = new int [27][6];
	private int sum = 162;
	Scanner input = new Scanner(System.in);
	
	public void fillPlane() {
		GroupOfPassengers currentGroup = new GroupOfPassengers();
		while(this.sum > 0) {
			System.out.println("Press enter to add another group to the plane...");
			input.nextLine();
			currentGroup.setSize();
			arrangeSeats(currentGroup.getSize());
			System.out.println("Added group of " + currentGroup.getSize() + " people...");
		}
	}
	
	public void arrangeSeats(int group) {
		switch(group) {
		case 1:
			addOnePerson();
		case 2:
			addTwoPeople();
		case 3:
			addThreePeople();
		}
	}
	
	public void addOnePerson() {
		if(sum - 1 >= 0) {
			for(int i = 0; i < 27; i++) {
				for(int j = 0; j < 6; j++) {
					if(capacity[i][j] == 0) {
						capacity[i][j] = 1;
						this.sum -= 1;
						return;
					}
				}
			}
		} else {
			System.out.println("Plane is already full...");
		}
	}
	
	public void addTwoPeople() {
		if(sum - 2 > 0) {
			for(int i = 0; i < 27; i++) {
				for(int j = 0; j < 6; j++) {
					switch(j) {
					case 0:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0) {
							capacity[i][j] = 2;
							capacity[i][j + 1] = 2;
							this.sum -= 2;
							return;
						}
					case 1:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0) {
							capacity[i][j] = 2;
							capacity[i][j + 1] = 2;
							this.sum -= 2;
							return;
						}
					case 3:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0) {
							capacity[i][j] = 2;
							capacity[i][j + 1] = 2;
							this.sum -= 2;
							return;
						}
					case 4:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0) {
							capacity[i][j] = 2;
							capacity[i][j + 1] = 2;
							this.sum -= 2;
							return;
						}
					}
				}
			}
		} else {
			System.out.println("Not enough seats for 2 more passengers...");
		}
	}
	
	public void addThreePeople() {
		if(sum - 3 > 0) {
			for(int i = 0; i < 27; i++) {
				for(int j = 0; j < 6; j++) {
					switch(j) {
					case 0:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0 && capacity[i][j + 2] == 0) {
							capacity[i][j] = 3;
							capacity[i][j + 1] = 3;
							capacity[i][j + 2] = 3;
							this.sum -= 3;
							return;
						}
					case 3:
						if(capacity[i][j] == 0 && capacity[i][j + 1] == 0 && capacity[i][j + 2] == 0) {
							capacity[i][j] = 3;
							capacity[i][j + 1] = 3;
							capacity[i][j + 2] = 3;
							this.sum -= 3;
							return;
						}
					}
				}
			}
		} else {
			System.out.println("Not enough seats for 3 more passengers...");
		}
	}
	
	public int getSum() {
		return this.sum;
	}
	
	public void displaySeats() {
		System.out.println("Press enter to display the seats...");
		input.nextLine();
		for(int i = 0; i < 27; i++) {
			for(int j = 0; j < 6; j++) {
				System.out.print(capacity[i][j]);
			}
			System.out.println();
		}
	}
}

File: class23_homework/B-Bozhidar-Nikolov-5/ruby_version/passenger_plane.rb
require_relative "group_of_passengers"

class PassengerPlane

	attr_accessor :capacity
	attr_accessor :sum
	attr_accessor :current_group

	def initialize
		self.capacity = Array.new(27) { Array.new(6, 0) }
		self.sum = 162
		self.current_group = GroupOfPassengers.new
	end

	def fill_plane
		self.current_group = GroupOfPassengers.new
		while self.sum > 0 
			puts "Press enter to add another group to the plane..."
			gets
			self.current_group.set_size
			arrange_seats self.current_group.size
			puts "Added group of #{self.current_group.size} people..."
		end
	end

	def arrange_seats group
		case group
		when 1 then
			add_one_person
		when 2 then
			add_two_people
		when 3 then
			add_three_people
		end
	end

	def add_one_person
		if self.sum - 1 >= 0
			for i in 0..self.capacity.length - 1
				for j in 0..self.capacity[0].length - 1
					if self.capacity[i][j] == 0
						self.capacity[i][j] = 1
						self.sum = self.sum - 1
						return
					end
				end
			end
		else
			puts "Plane is already full..."
		end
	end

	def add_two_people
		if self.sum - 2 >= 0
			for i in 0..self.capacity.length - 1
				for j in 0..self.capacity[0].length - 1
					case j
					when 0 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0
                                                        self.capacity[i][j] = 2
                                                        self.capacity[i][j + 1] = 2
                                                        self.sum = self.sum - 2
                                                        return
                                                end
					when 1 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0
                                                        self.capacity[i][j] = 2
                                                        self.capacity[i][j + 1] = 2
                                                        self.sum = self.sum - 2
                                                        return
                                                end
					when 3 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0
                                                        self.capacity[i][j] = 2
                                                        self.capacity[i][j + 1] = 2
                                                        self.sum = self.sum - 2
                                                        return
                                                end
					when 4 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0
                                                        self.capacity[i][j] = 2
                                                        self.capacity[i][j + 1] = 2
                                                        self.sum = self.sum - 2
                                                        return
                                                end
					end
				end
			end
		else
			puts "Not enough seats for 2 more passengers..."
		end
	end

	def add_three_people
		if self.sum - 3 >= 0
			for i in 0..self.capacity.length - 1
				for j in 0..self.capacity[0].length - 1
					case j
					when 0 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0 && self.capacity[i][j + 2] == 0
                                                        self.capacity[i][j] = 3
                                                        self.capacity[i][j + 1] = 3
                                                        self.capacity[i][j + 2] = 3
                                                        self.sum = self.sum - 3
                                                        return
                                                end
					when 3 then
						if self.capacity[i][j] == 0 && self.capacity[i][j + 1] == 0 && self.capacity[i][j + 2] == 0
                                                        self.capacity[i][j] = 3
                                                        self.capacity[i][j + 1] = 3
                                                        self.capacity[i][j + 2] = 3
                                                        self.sum = self.sum - 3
                                                        return
                                                end
					end
				end
			end
		else
			puts "Not enough seats for 3 more passengers..."
		end
	end

	def display_seats
		puts "Press enter to display seats..."
		gets
		for i in 0..self.capacity.length - 1
			for j in 0..self.capacity[0].length - 1
				print self.capacity[i][j]
			end
			puts
		end
	end
					
end

